src/xenia/hid/input_system.cc

```diff
@@ -62,10 +64,14 @@ X_RESULT InputSystem::GetState(uint32_t user_index, X_INPUT_STATE* out_state) {
 X_RESULT InputSystem::SetState(uint32_t user_index,
                                X_INPUT_VIBRATION* vibration) {
   SCOPE_profile_cpu_f("hid");
-
+  X_INPUT_VIBRATION* modified_vibration = new X_INPUT_VIBRATION();
```

Comment:
This is a memory leak 🥲
Just do `X_INPUT_VIBRATION modified_vibration;` to allocate it on the stack and pass `&modified_vibration`

Reply:
Unless you're already adding the scale for the vibration, you can just `return` immediately from the function, by the way — you don't need to set vibration to 0 if you never set it to anything other than 0 anyway.

Reply:
Smack me with a hammer for my stupidity... I dunno what I was thinking and why I thought it will be fine

Reply:
Well, if this var can someday be changed from gui and controller is currently rumbling, itll never stop iIrc

Reply:
The GUI button can explicitly stop the current vibration.